### PR TITLE
PlatformView: recreate surface if the controller changes

### PR DESCRIPTION
**Pull request by tauu**
## Description

Currently the surface of a platform view is only created only one when the state of PlatformViewLink is created. When the PlatformViewLink widget changes, the PlatformViewController in the corresponding state ist also updated. Just the surface is not updated even though it depends on the controller.

This PR changes this behaviour to recreate the surface whenever the controller ist updated.

## Tests

I added the following tests: none

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 1]
```diff
@@ -846,6 +846,9 @@ class _PlatformViewLinkState extends State<PlatformViewLink> {
 
     if (widget.viewType != oldWidget.viewType) {
       _controller?.dispose();
+      // the _surface has to be recreated as its controller is disposed
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/platform_view.dart]: nit: please use complete sentences starting with a capital letter and ending with a `.`.

**Review by goderbauer — COMMENTED** [review]
This will need a test.

/cc @amirh or @cyanglaz for detailed review.

**Review by amirh — CHANGES_REQUESTED** [review]
Thanks!

The change looks good, can you please add a unit test that verifies this scenario?

**Comment by tauu** [comment]
OK, I will try to provide a unit test. :-)

**Comment by goderbauer** [comment]
@tauu Do you still plan to come back to this soon-ish?

File: packages/flutter/lib/src/widgets/platform_view.dart (hunk 1 above)
Comment (tauu) [reply packages/flutter/lib/src/widgets/platform_view.dart]: done :-)

**Comment by tauu** [comment]
@goderbauer I am sorry for the delay. :-( Migrating from the flutter_web fork to the unforked flutter as well as incorporating a custom flutter build (with this PR) into the build system of my team took longer than anticipated.

The last commit extends the existing test "PlatformViewLink re-initializes when view type changes" to also verify that the surfaceFactory of a PlatformViewLink is called when the view type changes. Without this PR the extended test fails, with it the test succeeds.

**Review by cyanglaz — COMMENTED** [review]
tests LGTM.

**Review by amirh — APPROVED** [review]
LGTM

**Pull request merged**